### PR TITLE
Iet driver

### PR DESCRIPTION
**Pull request by chemistry-sourabh**
Issue #31

### CODE REVIEW

File: ims/einstein/iscsi/tgt/driver.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        except (IOError, OSError) as e:
+            raise iscsi_exceptions.TargetDeletionFailed(str(e))
+        except subprocess.CalledProcessError as e:
+            raise iscsi_exceptions.TargetDeletionFailed(str(e))
+
+    @log
+    def list_targets(self):
+        '''
+        Lists all the targets available. This queries tgt-admin and gets the
+        list of targets
```
Comment (naved001) [inline ims/einstein/iscsi/tgt/driver.py]: Blank line between the summary and the rest of the doscstring. PEP 257 it is.

File: ims/einstein/iscsi/iet/driver.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import ims.exception.config_exceptions as config_exceptions
+import ims.exception.iscsi_exceptions as iscsi_exceptions
+from ceph import RBD
+from ims.common.log import create_logger, log
+from ims.interfaces.iscsi import ISCSI
+
+logger = create_logger(__name__)
+
+
+# Not Tested so beware!!
```
Comment (naved001) [inline ims/einstein/iscsi/iet/driver.py]: How do we go about testing this?

Comment (chemistry-sourabh) [inline ims/einstein/iscsi/iet/driver.py]: Use an IET Iscsi server. The only one we have is being used by sec cloud so didn't want to touch it. If @sirushtim or @naved001 can write a script for installing let also we can add it to our CI.

**Comment by naved001** [comment]
For people who wants to review this and are overwhelmed by the size of it, the following files only contain pep8 fixes, so you could just skim through those.  
```
	test_ims.py
	test_rest.py	
	test_operations.py
	einstein/__inint__.py
	test_project.py
	test_image.py
	database/__init__.py
	haas_exceptions.py
	file_system_exceptions.py
	project.py
	snapshot.py
	image.py
	cli.py
```

Also, we still have to review the main meaty part of this PR (the actual driver related stuff)

**Comment by chemistry-sourabh** [comment]
Just review the commit which is not present in the previous PR.

**Comment by chemistry-sourabh** [comment]
For your status errors.
These status errors will be seen by the end user and they need to reflect the already established HTTP errors. Since ISCSI is internal to BMI it will be an internal error hence 500.

What about adding error codes in the message like software errors are shown in windows ?
For example Failed to create target (EC5001)
@CCI-MOC/bmi-breakers what do you guys think ?

**Comment by chemistry-sourabh** [comment]
Review commit 9a9aad1 and e344bf8 of this PR. other commits can be reviewed in previous PRs

**Comment by naved001** [comment]
@chemistry-sourabh do you want to close this PR and the generic driver loader one? because there are a lot of merge conflicts. And since we are discussing the redesign at this point, you can reopen it after that. What do you think?

**Comment by naved001** [comment]
closing this for now until we decide how to redesign BMI.

**Pull request closed without merging**